presto-spi/src/main/java/io/prestosql/spi/expression/ConnectorExpressionVisitor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.expression;
+
+import javax.annotation.Nullable;
+
+public abstract class ConnectorExpressionVisitor<R, C>
```

Comment:
I intentionally avoided adding a visitor abstraction for ConnectorExpression, as it makes the hierarchy rigid. We'll probably want to make it extensible in the future as we start expanding on the kinds of things that can be communicated across the SPI (e.g.., connector-provided expression types and rules)